api/v1alpha1/system_types.go

```diff
@@ -0,0 +1,793 @@
+		SuccessThreshold:    pointer.Int32Ptr(1),
+		FailureThreshold:    pointer.Int32Ptr(5),
+	}
+	systemDefaultSidekiqPDB defaultPodDisruptionBudgetSpec = defaultPodDisruptionBudgetSpec{
+		MaxUnavailable: util.IntStrPtr(intstr.FromInt(1)),
+	}
+
+	// Sphinx
+	systemDefaultSphinxDeltaIndexInterval  int32                           = 5
+	systemDefaultSphinxFullReindexInterval int32                           = 60
+	systemDefaultSphinxPort                int32                           = 9306
+	systemDefaultSphinxBindAddress         string                          = "0.0.0.0"
+	systemDefaultSphinxConfigFile          string                          = "/opt/system/db/sphinx/preview.conf"
+	systemDefaultSphinxDBPath              string                          = "/opt/system/db/sphinx"
+	systemDefaultDatabaseStorageSize       string                          = "30Gi"
+	systemDefaultSphinxPIDFile             string                          = "/opt/system/tmp/pids/searchd.pid"
+	systemDefaultSphinxStorage             string                          = "30Gi"
+	systemDefaultSphinxResources           defaultResourceRequirementsSpec = defaultResourceRequirementsSpec{
+		Requests: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("250m"),
+			corev1.ResourceMemory: resource.MustParse("4Gi"),
+		},
+		Limits: corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("750m"),
+			corev1.ResourceMemory: resource.MustParse("5Gi"),
+		},
+	}
+	systemDefaultSphinxLivenessProbe defaultProbeSpec = defaultProbeSpec{
+		InitialDelaySeconds: pointer.Int32Ptr(60),
+		TimeoutSeconds:      pointer.Int32Ptr(3),
+		PeriodSeconds:       pointer.Int32Ptr(15),
+		SuccessThreshold:    pointer.Int32Ptr(1),
+		FailureThreshold:    pointer.Int32Ptr(5),
+	}
+	systemDefaultSphinxReadinessProbe defaultProbeSpec = defaultProbeSpec{
+		InitialDelaySeconds: pointer.Int32Ptr(60),
+		TimeoutSeconds:      pointer.Int32Ptr(5),
+		PeriodSeconds:       pointer.Int32Ptr(30),
+		SuccessThreshold:    pointer.Int32Ptr(1),
+		FailureThreshold:    pointer.Int32Ptr(5),
+	}
+)
+
+// SystemSpec defines the desired state of System
+type SystemSpec struct {
+	// Application specific configuration options for System components
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	Config SystemConfig `json:"config"`
+	// Image specification for the component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Image *ImageSpec `json:"image,omitempty"`
+	// Application specific configuration options
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	App *SystemAppSpec `json:"app,omitempty"`
+	// Sidekiq specific configuration options
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Sidekiq *SystemSidekiqSpec `json:"sidekiq,omitempty"`
+	// Sphinx specific configuration options
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Sphinx *SystemSphinxSpec `json:"sphinx,omitempty"`
```

Comment:
I don't really see the point in adding this. We will never have sphinx in an external instance (if we ever have it, let's implement it when required). As for the resources thing, you can just tune down the resource requirements. In fact we should always deploy the sphinx component, even in dev, because otherwise how do you test that it works?

Reply:
well, we did that because sphinx was consuming most of the resources of our Dev clusters (although was not needed for app to work), if we can leave with that, by maybe just set the replicas to 0, for me is fair enough